src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
 
         /// <summary>
         /// Invoked before redirecting to the identity provider to sign out.
         /// </summary>
-        Task RedirectToEndSessionEndpoint(RedirectContext context);
+        Task RedirectToIdentityProviderForSignOut(RedirectContext context);
 
         /// <summary>
         /// Invoked after "authorization code" is redeemed for tokens at the token endpoint.
         /// </summary>
-        Task TokenResponseReceived(TokenResponseReceivedContext context);
+        Task TokenResponseMessageReceived(TokenResponseMessageReceivedContext context);
```

Comment:
@HaoK see, I think that your comment makes exactly my point. This event is not signaling that we received a token - it is signaling that we received a message from the token endpoint. The kind of extensions a developer can inject here might have nothing to do with the token itself. Initially I suggested that here we should simply raise another MessageReceived, and have something in the event parameters to signal the nature of the message - but @Tratcher convinced me otherwise. This kind of message is specific enough to the OIDC flows that, like the authorizationcodereceived, it deserves a protocol specific name (whereas MessageReceived, meant as a frontchannel auth message, is an artifact that is common to all redirect based protocols).
Note, I am not saying that names length isn't important - but unless we end up in the realm of the absurd (exhibit A: usewindowsactivedirectorybearerauthentication) I would consider clarity the highest order bit. Perhaps we have different threshold for absurd, tho :) do you still feel strongly against adding "message" after having read the above?